Conform to latest Ruff import order

With the latest version of ruff (0.6.4) the cookiecutter template gives a "I001 [*] Import block is un-sorted or un-formatted" error.
For example in the parser test:
```sh
tests/parsers/test_parser.py:1:1: I001 [*] Import block is un-sorted or un-formatted
  |
1 | / import logging
2 | | 
3 | | from nomad.datamodel import EntryArchive
4 | | from plugin_with_example.parsers.parser import NewParser
5 | | 
6 | | 
7 | | def test_parse_file():
  | |_^ I001
8 |       parser = NewParser()
9 |       archive = EntryArchive()
  |
  = help: Organize imports
```
But also in other places. Nothing urgent as it's easily fixed with `ruff check . --fix` but it would be nice if the pipeline doesn't fail directly.

--- a/.github/workflows/run-checks.yml
+++ b/.github/workflows/run-checks.yml
@@ -24,6 +24,9 @@ jobs:
       - uses: actions/setup-python@v5
         with:
           python-version: 3.11
+      - name: Install Homebrew libmagic
+        if: runner.os == 'macOS'
+        run: brew install libmagic
       - name: Install uv
         uses: astral-sh/setup-uv@v2
       - name: Run tox

--- a/.github/workflows/run-checks.yml
+++ b/.github/workflows/run-checks.yml
@@ -14,13 +14,18 @@ on:
 
 jobs:
   tox:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-14]
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
         with:
-          python-version: 3.9
-      - name: Install tox
-        run: pip install tox-uv
+          python-version: 3.11
+      - name: Install uv
+        uses: astral-sh/setup-uv@v2
       - name: Run tox
-        run: tox -e py
+        run: |
+          uv run --with pytest --with pytest-cookies --with tox-uv pytest

--- a/tests/test_create_template.py
+++ b/tests/test_create_template.py
@@ -10,13 +10,11 @@ def run_tox(plugin):
     shutil.copy("tox.ini", tox_plugin_path)
     command = [
         "tox",
-        "-v",
         "--workdir",
         plugin,
         "-c",
         tox_plugin_path,
-        "-e",
-        "py",
+        "run",
     ]
 
     process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
@@ -25,9 +23,10 @@ def run_tox(plugin):
     print("Command Output:")
     print(stdout.decode())
     print(stderr.decode())
+    assert process.returncode == 0
 
 
-@pytest.mark.parametrize("plugin_name", ["foo-bar"])
+@pytest.mark.parametrize("plugin_name", ["long-name-foo-bar"])
 @pytest.mark.parametrize(
     "include_normalizer",
     [True, False],
@@ -113,4 +112,5 @@ def test_run_cookiecutter_and_plugin_tests(
             "parsers",
         ).is_dir()
 
-    run_tox(str(result.project_path))
+    if include_app and include_parser and include_normalizer and include_schema_package:
+        run_tox(str(result.project_path))

--- a/tox.ini
+++ b/tox.ini
@@ -1,18 +1,34 @@
 [tox]
 recreate = true
-skipsdist = true
-envlist = py39
+skipsdist = false
+env_list =
+    py311
+    lint
+    format
 
-[testenv]
+[testenv:py311]
 setenv = 
     UV_EXTRA_INDEX_URL=https://gitlab.mpcdf.mpg.de/api/v4/projects/2187/packages/pypi/simple
 
 deps = pytest
        pytest-cookies
        tox
-       nomad-lab>=1.3.0
+       nomad-lab>=1.3.6
 
 commands = pytest {posargs:tests}
+
+
+[testenv:lint]
+description = run linters
+skip_install = true
+deps = ruff
+commands = ruff check src tests
+
+[testenv:format]
+description = run format
+skip_install = true
+deps = ruff
+commands = ruff format --check src tests
 
 [pytest]
 norecursedirs = "{{cookiecutter.plugin_name}}"

--- a/{{cookiecutter.plugin_name}}/py_sources/src/apps/__init__.py
+++ b/{{cookiecutter.plugin_name}}/py_sources/src/apps/__init__.py
@@ -1,7 +1,6 @@
 from nomad.config.models.plugins import AppEntryPoint
 from nomad.config.models.ui import App, Column, Columns, FilterMenu, FilterMenus
 
-
 app_entry_point = AppEntryPoint(
     name='NewApp',
     description='New app entry point configuration.',

--- a/{{cookiecutter.plugin_name}}/py_sources/src/parsers/parser.py
+++ b/{{cookiecutter.plugin_name}}/py_sources/src/parsers/parser.py
@@ -11,9 +11,8 @@ if TYPE_CHECKING:
     )
 
 from nomad.config import config
-from nomad.parsing.parser import MatchingParser
-
 from nomad.datamodel.metainfo.workflow import Workflow
+from nomad.parsing.parser import MatchingParser
 
 configuration = config.get_plugin_entry_point(
     '{{cookiecutter.module_name}}.parsers:parser_entry_point'

--- a/{{cookiecutter.plugin_name}}/py_sources/src/schema_packages/__init__.py
+++ b/{{cookiecutter.plugin_name}}/py_sources/src/schema_packages/__init__.py
@@ -7,7 +7,7 @@ class NewSchemaPackageEntryPoint(SchemaPackageEntryPoint):
 
     def load(self):
         from {{cookiecutter.module_name}}.schema_packages.schema_package import m_package
- 
+
         return m_package
 
 

--- a/{{cookiecutter.plugin_name}}/py_sources/tests/apps/test_app.py
+++ b/{{cookiecutter.plugin_name}}/py_sources/tests/apps/test_app.py
@@ -3,4 +3,3 @@ def test_importing_app():
     from {{cookiecutter.module_name}}.apps import app_entry_point
 
     assert app_entry_point.app.label == 'NewApp'
-

--- a/{{cookiecutter.plugin_name}}/py_sources/tests/parsers/test_parser.py
+++ b/{{cookiecutter.plugin_name}}/py_sources/tests/parsers/test_parser.py
@@ -1,6 +1,7 @@
 import logging
 
 from nomad.datamodel import EntryArchive
+
 from {{cookiecutter.module_name}}.parsers.parser import NewParser
 
 

--- a/{{cookiecutter.plugin_name}}/pyproject.toml
+++ b/{{cookiecutter.plugin_name}}/pyproject.toml
@@ -35,7 +35,10 @@ maintainers = [
     { name = "{{cookiecutter.full_name}}", email = "{{cookiecutter.email}}" },
 ]
 license = { file = "LICENSE" }
-dependencies = ["nomad-lab>=1.3.0"]
+dependencies = [
+    "nomad-lab>=1.3.0",
+    "python-magic-bin; sys_platform == 'win32'",
+]
 
 [project.urls]
 Repository = "https://github.com/{{cookiecutter.github_username}}/{{cookiecutter.plugin_name}}"

